scripts/pre_commit_linter.py

```diff
@@ -471,6 +466,54 @@ def _get_expression_from_node_if_one_exists(
     return expression
 
 
+def _walk_with_gitignore(root, exclude_dirs):
+    """A walk function similar to os.walk but this would ignore the files and
+    directories which is not tracked by git. Also, this will ignore the
+    directories mentioned in exclude_dirs.
+
+    Args:
+        root: str. The path from where the function should start walking.
+        exclude_dirs: list(str). A list of dir path which should to be ignored.
+
+    Yields:
+        str. list(str). A list of unignored files.
+    """
+    dirs, nondirs = [], []
```

Comment:
Does nondirs mean filepaths? If so perhaps that would be a better name.

Reply:
Done!